### PR TITLE
Upgrade to cilium v1.12.2 and v1.11.9

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note updates
Upgrade to cilium v1.12.2 and v1.11.9
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
/hold
Dualstack is broken after this guide

File: addons/cilium/cilium_v1.12.yaml [hunk 1]
```diff
@@ -1069,6 +1074,7 @@ spec:
         - name: cilium-config-path
           mountPath: /tmp/cilium/config-map
           readOnly: true
+      terminationMessagePolicy: FallbackToLogsOnError
```
Comment (rastislavs) [inline addons/cilium/cilium_v1.12.yaml]: Wrong indentation - please add 2 extra spaces in front of the `terminationMessagePolicy`

**Pull request merged**